console/metrics.md

```diff
@@ -0,0 +1,174 @@
+---
+title: Console Metrics for Telemetry
+authors:
+  - "@benjaminapetersen"
+reviewers:
+  - "@spadgett"
+  - "@bparees"
+approvers:
+  - "@spadgett"
+  - "@bparees"
+creation-date: 2019-09-20
+last-updated: 2019-09-20
+status: implementable
+---
+
+# Console Metrics for Telemetry
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+
+The purpose of this enhancement is to enable a set of telemetry metrics 
+for the console in order to improve the support of our customer environments.  
+
+## Motivation
+
+### Goals
+
+- Tracking metrics via telemetry would allow us to gather feature usage data for the 
+console so that we can focus development efforts on the features customers find most 
+valuable.
+
+### Non-Goals
+
+- Analytics will not be handled in this proposal.  Previously, the console supported an optional 
+Google Analytics identifier. No other analytics platform is or has been supported.
+- Error tracking will not be handled in this proposal.  We have discussed the use of tools like 
+Sentry for collecting errors and stack traces to improve our ability to understand console 
+error frequency and severity within the browser.
+
+## Proposal
+
+- Track the console URL as a metadata metric like `console_url` to support UHC to provide backlinking 
+to the console.
+- Track API requests from the console to the API server as a metric like `console_api_requests_total` 
+with labels such as `referer` and `path` to clearly understand how the console consumes the API and 
+make the correct design and optimizations for it.
+- Track console pageviews as a metric like `console_page_view_total` to understand what pages and features
+customers find most useful.
+- Track user agent as a metric like `console_user_agent_total` to understand which browsers are most used 
+by customers.  This metric may be collapsed into `console_page_view_total` with the `user agent` tracked 
+as an additional label. 
+- Track console login as a metric like `console_login_total` to understand average console use.
```

Comment:
wouldn't this be part of page view total?  (how many times was the login page viewed?)

Reply:
Agree, we really don't need to split this out as a separate metric.